File: ios/Models/OverlayElement.swift
import Foundation

struct OverlayElement: Codable, Identifiable {
    enum ElementType: String, Codable {
        case distanceMarker
        case layupMarker
        case targetLine
        case windHint
        case safetyBanner
        case offlineBadge
        case perfOverlay
    }

    let id: UUID
    let sessionID: UUID
    let anchorID: UUID
    let type: ElementType
    var distanceMeters: Double?
    var windTier: String?
    var isVisible: Bool
}

File: android/app/src/main/java/com/golfiq/hud/model/OverlayElement.kt
package com.golfiq.hud.model

import java.util.UUID

data class OverlayElement(
    val id: UUID,
    val sessionId: UUID,
    val anchorId: UUID,
    val type: ElementType,
    val distanceMeters: Double?,
    val windTier: String?,
    val isVisible: Boolean,
) {
    enum class ElementType {
        DISTANCE_MARKER,
        LAYUP_MARKER,
        TARGET_LINE,
        WIND_HINT,
        SAFETY_BANNER,
        OFFLINE_BADGE,
        PERF_OVERLAY
    }
}